validator/sawtooth_validator/server/cli.py

```diff
@@ -184,53 +90,26 @@ def load_validator_config(first_config, config_dir):
         configs=[first_config, toml_config, default_validator_config])
 
 
-def create_validator_config(opts):
-    bind_network = None
-    bind_component = None
-    if opts.bind:
-        for bind in opts.bind:
-            if "network" in bind:
-                bind_network = bind[bind.find(":") + 1:]
-            if "component" in bind:
-                bind_component = bind[bind.find(":") + 1:]
-    return ValidatorConfig(
-        bind_network=bind_network,
-        bind_component=bind_component,
-        endpoint=opts.endpoint,
-        peering=opts.peering,
-        seeds=opts.seeds,
-        peers=opts.peers,
-        scheduler=opts.scheduler,
-        roles=opts.network_auth,
-        opentsdb_url=opts.opentsdb_url,
-        opentsdb_db=opts.opentsdb_db,
-        minimum_peer_connectivity=opts.minimum_peer_connectivity,
-        maximum_peer_connectivity=opts.maximum_peer_connectivity
-    )
-
-
-def main(args=None):
-    if args is None:
-        prog_name = sys.argv[0]
-        args = sys.argv[1:]
-    else:
-        prog_name = args[0]
-        args = args[1:]
-    opts = parse_args(prog_name, args)
-    verbose_level = opts.verbose
+class _Opts:
+    def __init__(self, opts_string):
+        opts = json.loads(opts_string)
 
-    # Determine if any args which support delimited lists should be
-    # modified
-    if opts.peers:
-        opts.peers = _split_comma_append_args(opts.peers)
+        for key, val in opts.items():
+            setattr(self, key, val)
 
-    if opts.seeds:
-        opts.seeds = _split_comma_append_args(opts.seeds)
+        self.network_public_key = None
+        self.network_private_key = None
+        self.permissions = None
+        self.opentsdb_username = None
+        self.opentsdb_password = None
 
-    init_console_logging(verbose_level=verbose_level)
 
-    if opts.network_auth:
-        opts.network_auth = {"network": opts.network_auth}
+def main(opts_string):
+    opts = _Opts(opts_string)
```

Comment:
Reducing this to arbitrary key/value pairs in not acceptable as it removes the definition of the object. This would be a regression to what we had in 0.7.